go-controller/pkg/ovn/address_set/address_set.go

```diff
@@ -74,6 +77,49 @@ func (asf *ovnAddressSetFactory) NewAddressSet(name string, ips []net.IP) (Addre
 	return res, nil
 }
 
+// EnsureAddressSet ensures the address_set with the given name exists and if it does not creates an empty addressSet
+func (asf *ovnAddressSetFactory) EnsureAddressSet(name string) error {
+	hashedAddressSetNames := []string{}
```

Comment:
isnt the body of this function the same as calling 
```
func newOvnAddressSet(name string, ips []net.IP) (*ovnAddressSet, error) {
```
with an empty IP slice? or cant you just use NewAddressSet with an empty slice?

Reply:
it is not the same as `newOvnAddressSet()` or calling `NewAddressSet()` with an empty slice the problem with that is they both would call `as.setIPs()` on the empty slice which would clear the IP addresses from any existing address sets.

Reply:
oh. It looks like if the address set already exists that is the case:

```
	if uuid != "" {
		klog.V(5).Infof("New(%s) already exists; updating IPs", asDetail(as))
		// ovnAddressSet already exists in the database; just update IPs
		if err := as.setIPs(ips); err != nil {
			return nil, err
		}
	} else {
		// ovnAddressSet has not been created yet. Create it.
		args := []string{
			"create",
			"address_set",
			"name=" + as.hashName,
			"external-ids:name=" + as.name,
		}
		joinedIPs := joinIPs(as.allIPs())
		if len(joinedIPs) > 0 {
			args = append(args, "addresses="+joinedIPs)
		}
		as.uuid, stderr, err = util.RunOVNNbctl(args...)
		if err != nil {
			return nil, fmt.Errorf("failed to create address set %q, stderr: %q (%v)",
				asDetail(as), stderr, err)
		}
	}
```

I would just modify that function to be addOrUpdateAddressSet, and check the len(ips) >0  before it tries to call setIPs.

Reply:
there is a test case specifically that calling `NewAddressSet()` with an existing addressSet and no IPs clears the ip address from the address set I am not sure why but do we want to change that?

Reply:
ok lets just leave it as is and we can consolidate later.